exercises/concept/wizards-and-warriors/src/test/java/FighterTest.java

```diff
@@ -1,87 +1,217 @@
-    public void testWizardsDamagePointsAfterPreparingSpell() {
-        Wizard wizard = new Wizard();
-        Warrior warrior = new Warrior();
-        wizard.prepareSpell();
-        assertThat(wizard.damagePoints(warrior)).isEqualTo(12);
+    @DisplayName("The getDamagePoints method of the Warrior returns 6 when target is not vulnerable")
+    void testWarriorsDamagePointsWhenTargetNotVulnerable() {
+        assertThat(warrior.getDamagePoints(new InvulnerableFighter())).isEqualTo(6);
     }
 
     @Test
     @Tag("task:5")
-    @DisplayName("The damagePoints method of the Warrior returns 10 when target is vulnerable")
-    public void testWarriorsDamagePointsWhenTargetVulnerable() {
-        Warrior warrior = new Warrior();
-        Wizard wizard = new Wizard();
-        assertThat(warrior.damagePoints(wizard)).isEqualTo(10);
+    @DisplayName("Test that the Wizard class exists in the code")
+    void testWizardClassExists() {
+        try {
+            Class.forName("Wizard");
+        } catch (ClassNotFoundException e) {
+            fail("Should have a class called Wizard");
+        }
     }
 
     @Test
     @Tag("task:5")
-    @DisplayName("The damagePoints method of the Warrior returns 6 when target is not vulnerable")
-    public void testWarriorsDamagePointsWhenTargetNotVulnerable() {
-        Warrior warrior = new Warrior();
-        Wizard wizard = new Wizard();
+    @DisplayName("Test that the Wizard class is an instance of the Fighter class")
+    void testWizardIsInstanceOfFighter() throws ClassNotFoundException {
+        assertThat(Class.forName("Wizard")).isAssignableTo(Fighter.class);
+    }
+
+    @Test
+    @Tag("task:6")
+    @DisplayName("Wizard has the toString method implemented")
+    void implemented_to_string_for_wizard() {
+        assertThat(wizard.hasMethod("toString"))
+                .withFailMessage("Method toString must be created")
+                .isTrue();
+        assertThat(wizard.isMethodReturnType(String.class, "toString"))
+                .withFailMessage("Method toString must return a String")
+                .isTrue();
+    }
+
+    @Test
+    @Tag("task:6")
+    @DisplayName("The toString method of the Wizard returns the correct description of the fighter")
+    void testWizardToString() {
+        assertThat(wizard.toString()).isEqualTo("Fighter is a Wizard");
+    }
+
+    @Test
+    @Tag("task:7")
+    @DisplayName("Wizard has the prepareSpell method implemented")
+    void implemented_prepare_spell_for_wizard() {
+        assertThat(wizard.hasMethod("prepareSpell"))
+                .withFailMessage("Method prepareSpell must be created")
+                .isTrue();
+    }
+
+    @Test
+    @Tag("task:7")
+    @DisplayName("The prepareSpell method makes the Wizard not vulnerable")
+    void testWizardVulnerable() {
         wizard.prepareSpell();
-        assertThat(warrior.damagePoints(wizard)).isEqualTo(6);
+        assertThat(wizard.isVulnerable()).isFalse();
+    }
```

Comment:
Hmm it's a bit strange that this test requires the `isVulnerable` method to be implemented correctly, but that is part of the next task. Maybe tasks 7 and 8 should be merged into one? So adding the `prepareSpell()` method and overriding the `isVulnerable()` method are bundled into a single task? This might make sense as they are very much related.

Reply:
Sure, I agree!

Reply:
Done!